subprojects/enterprise/src/main/java/org/gradle/internal/enterprise/GradleEnterprisePluginBackgroundJobExecutor.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.enterprise;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.Executor;
+import java.util.concurrent.RejectedExecutionException;
+
+/**
+ * Runs submitted jobs in the background. The implementation is provided by Gradle.
+ * The intended use is to provide workers for the {@code buildScan.background} callbacks.
+ * <p>
+ * The instances of this interface are thread-safe.
+ */
+public interface GradleEnterprisePluginBackgroundJobExecutor extends Executor {
```

Comment:
It might be better to has-a Executor than is-a Executor, for example so that we can later offer different executors with different semantics, or even not use an Executor at all.